File: auth/src/main/java/com/firebase/ui/auth/data/model/FirebaseAuthUIAuthenticationResult.java
package com.firebase.ui.auth.data.model;

import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract;
import com.firebase.ui.auth.IdpResponse;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Result of launching a {@link FirebaseAuthUIActivityResultContract}
 */
public class FirebaseAuthUIAuthenticationResult {

    @Nullable
    private final IdpResponse idpResponse;
    @NonNull
    private final Integer resultCode;

    public FirebaseAuthUIAuthenticationResult(@NonNull Integer resultCode, @Nullable IdpResponse idpResponse) {
        this.idpResponse = idpResponse;
        this.resultCode = resultCode;
    }

    /**
     * The contained {@link IdpResponse} returned from the Firebase library
     */
    @Nullable
    public IdpResponse getIdpResponse() {
        return idpResponse;
    }

    /**
     * The result code of the received activity result
     *
     * @see android.app.Activity.RESULT_CANCELED
     * @see android.app.Activity.RESULT_OK
     */
    @NonNull
    public Integer getResultCode() {
        return resultCode;
    }

    @Override
    public int hashCode() {
        int result = idpResponse == null ? 0 : idpResponse.hashCode();
        result = 31 * result + resultCode.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "FirebaseAuthUIAuthenticationResult{" +
                "idpResponse=" + idpResponse +
                ", resultCode='" + resultCode +
                '}';
    }
}

File: auth/src/main/java/com/firebase/ui/auth/data/model/FirebaseAuthUIAuthenticationResult.kt
/*
 * Copyright 2025 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the
 * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.firebase.ui.auth.data.model

import com.firebase.ui.auth.IdpResponse

/**
 * Result of launching a [FirebaseAuthUIActivityResultContract]
 */
data class FirebaseAuthUIAuthenticationResult(
    /**
     * The result code of the received activity result
     *
     * @see android.app.Activity.RESULT_CANCELED
     * @see android.app.Activity.RESULT_OK
     */
    val resultCode: Int,
    /**
     * The contained [IdpResponse] returned from the Firebase library
     */
    val idpResponse: IdpResponse?
)